library/core/src/main/java/com/google/android/exoplayer2/C.java

```diff
@@ -1073,6 +1073,10 @@ private C() {}
   /** Indicates the track contains a text that has been edited for ease of reading. */
   public static final int ROLE_FLAG_EASY_TO_READ = 1 << 13;
 
+  // TODO - not a 'role' in the sense it is parsed from the CHARACTERISTICS attribute... forced if iFrame only
+  /** Indicates the track is an IDR (IFrame) only track for trick play */
+  public static final int ROLE_FLAG_TRICK_PLAY = 1 << 14;
+
```

Comment:
I'm certainly not married to this idea...  But, there needs to be some hook in the `Format` to identify the Iframe only tracks.  It's unfortunate the HLS spec does not mandate (or even allow) specifying the frame rate on the Master Playlist EXT-X-IFRAME-ONLY